Add missing configuration for Bluetooth and storage partition for settings

ppryga-nordic opened the pull request:
There is missing default Kconfig configuration related with Bluetooth protocol for nRF54h20 DK.
There is missing storage partition configuration for settings subsystem for nRF54h20 DK.
There is missing nRF54H20 DK configuration in hci_ipc Bluetooth sample.

Add missing items.

ppryga-nordic committed:
boards: nrf54h20: Fix storage partitions for setting subsytem

There was missing storage partition that is required for setting
subsytem. Add missing entries to board files.

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>

ppryga-nordic committed:
board: nrf54h20dk: Add missing default BT related configs

There are default configs that are provided by default for
e.g. nRF5340dk to enable HCI IPC communication type for
application domain and enable BT controller for radio domain.
These configs were missing for nRF54h20 DK.

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>

ppryga-nordic commented:
@hubertmis  please check the commit with storage partition if that fits into planned memory map.

57300 commented on boards/nordic/nrf54h20dk/Kconfig.nrf54h20dk:
```diff
@@ -6,3 +6,18 @@ config BOARD_NRF54H20DK
+if BOARD_NRF54H20DK_NRF54H20_CPUAPP
+
+choice BT_HCI_BUS_TYPE
+	default BT_HCI_IPC if BT
```
Please create `Kconfig.defconfig` and move these changes there.

57300 replied:
Now revert all changes to `Kconfig.nrf54h20dk`.

57300 commented on boards/nordic/nrf54h20dk/nrf54h20dk_nrf54h20-memory_map.dtsi:
```diff
@@ -174,11 +174,26 @@
 
 		cpuapp_slot0_partition: partition@a6000 {
-			reg = <0xa6000 DT_SIZE_K(512)>;
+			reg = <0xa6000 DT_SIZE_K(488)>;
```
If this needs to change, then please update `nrf54h20dk_nrf54h20_cpuapp.yaml` as well.

hubertmis replied:
If my calculations are correct, the `cpuapp_rw_partitions` is placed 552 KB further than this partition. So there is no need to shrink it. It could be even extended to 552 KB if needed.

BTW, is there a reason for this 64kB hole between the regions?

57300 commented on boards/nordic/nrf54h20dk/nrf54h20dk_nrf54h20-memory_map.dtsi:
```diff
@@ -174,11 +174,26 @@
+		#size-cells = <1>;
+
+		storage_partition: partition@130000 {
+			label = "storage";
```
I don't think these labels are used anymore.

ppryga-nordic commented:
I'm wondering why compliance doesn't see KConfig symbols defined in Zephyr...

57300 commented:
> I'm wondering why compliance doesn't see KConfig symbols defined in Zephyr...

It's because `Kconfig.<board_name>` is part of the reusable board/SoC Kconfig subtree, introduced with HWMv2. This subtree is reused by sysbuild, so it must not contain settings that are undefined in sysbuild, and compliance can now check for this.

ppryga-nordic committed:
samples: bluetooth: hci_ipc: Add nrf54h20 DK board overlay

There are required DTS entries to nable peripherals required
for IPC communication between Radio and APP CPUs.

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>

ppryga-nordic commented:
@57300 requested change done. Fixed CI and compliance. @anangl  thank you for your help.

If there are no more objections then PR can be merged and settings should be working.

ppryga-nordic commented:
Added a commit with nRF54H20 board configuration for hci_ipc BLE sample.

anangl commented on boards/nordic/nrf54h20dk/nrf54h20dk_nrf54h20_cpuapp.yaml:
```diff
@@ -10,6 +10,6 @@ toolchain:
   - zephyr
 ram: 256
-flash: 512
+flash: 600
```
Why is this modified? `cpuapp_slot0_partition` size is still 512 kB.

ppryga-nordic replied:
Isn't it size of flash used ? @57300 asked to update it.

anangl replied:
No, it's the available flash. See [Board configuration](https://docs.zephyrproject.org/latest/develop/test/twister.html#board-configuration).

ppryga-nordic replied:
Ok then I'll revert it

ppryga-nordic replied:
done

ppryga-nordic commented:
@anangl change reverted.

ppryga-nordic commented:
@carlescufi  @nashif could on of you merge the PR?
